ocaml/xapi/xapi_pif.ml

```diff
@@ -767,6 +767,21 @@ let set_property ~__context ~self ~name ~value =
         Nm.bring_pif_up ~__context pif
     ) (self :: vlan_pifs)
 
+let pif_has_clustering_enabled ~__context (self : API.ref_PIF) =
+  let network_ref = Db.PIF.get_network ~__context ~self |> Ref.string_of in
+  (Db.Cluster.get_refs_where ~__context
+     ~expr:Db_filter_types.(Eq(Field "network", Literal network_ref)))
+  |> function
+  | []   -> false
+  | a::_ -> true
+
+let set_disallow_unplug ~__context ~self ~value =
+  if pif_has_clustering_enabled ~__context self
+  then
+    let network_ref = Db.PIF.get_network ~__context ~self |> Ref.string_of in
```

Comment:
We already got this from the DB in the function above, so this could be optimised a little. Not a big deal though.